From 0ec1bf38b82e3d945e094d0e9f33ef1dbee1918e Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 04 Jul 2018 18:13:39 +0300
Subject: [PATCH] install: Add ASIC_BIN_DIR for installing production binaries fixup

Now there is /tools/hl_shared_dir/production/embedded/head by default
and also /tools/hl_shared_dir/production/embedded/latest symbolic link
for non head i.e. ASIC_BIN_DIR provided at command line
---

diff --git a/Makefile b/Makefile
index 3447bc3..2db7396 100644
--- a/Makefile
+++ b/Makefile
@@ -110,7 +110,7 @@
 	@mkdir -p $@
 dali-pldm-install: ${PLDM_HEX_DIR}
 	-cp $(IMAGES_DIR)/dali-spi-flash.hex $</
-ASIC_BIN_DIR ?= /tools/hl_shared_dir/production/embedded/latest
+ASIC_BIN_DIR ?= /tools/hl_shared_dir/production/embedded/head
 ${ASIC_BIN_DIR}:
 	@mkdir -p $@
 dali-asic-install: ${ASIC_BIN_DIR}
@@ -118,6 +118,9 @@
 	-cp $(IMAGES_DIR)/dali-spi-flash.bin $</
 	-cp $(IMAGES_DIR)/dali-u-boot.bin $</
 	-cp $(IMAGES_DIR)/dali-u-boot-spl.bin $</
+ifneq ($(notdir $(realpath $(ASIC_BIN_DIR))), head)
+	ln -nfs $$(basename $(ASIC_BIN_DIR)) $$(dirname $(ASIC_BIN_DIR))/latest
+endif
 
 .PHONY: dali-distclean
 dali-distclean: $(OUTPUT_DIR)/.config
